server/src/analytics/GoogleAnalytics.ts

```diff
@@ -63,75 +55,39 @@ export class GoogleAnalytics implements Analytics {
 
   private _buildPayloadFrom(
     taskName: string,
-    machineId: string,
-    more?: RawAnalyticsPayload
+    machineId: string
   ): AnalyticsPayload {
-    const defaultAnalytics: DefaultRawAnalyticsPayload = {
-      // Measurement protocol version.
-      v: "1",
-
-      // Tracking Id.
-      tid: this.googleTrackingId,
-
-      // User agent, must be present.
-      // We use it to inform Node version used and OS.
-      // Example:
-      //   Node/v8.12.0 (Darwin 17.7.0)
-      ua: this._getUserAgent(),
-
-      // Client Id.
-      cid: machineId,
-
-      // Hit type, we're only using timing for now.
-      t: "pageview",
-
-      // Document page
-      dp: `/${taskName}`,
-
-      // Custom dimension 1: extension version
-      //	 Example: 'v1.0.0'.
-      cd1: `v${this.extensionVersion}`,
-
-      // Data source -> vs code or coc
-      ds: this.clientName,
+    return {
+      client_id: machineId,
+      user_id: machineId,
+      user_properties: {
+        extensionVersion: { value: this.extensionVersion },
+        languageClient: { value: this.clientName },
+        operatingSystem: { value: os.type() },
+      },
+      events: [
+        {
+          name: taskName,
+          params: {
+            engagement_time_msec: "10000",
+            session_id: machineId,
```

Comment:
Shouldn't this be a randomly generated id on startup?

Reply:
You are correct, forgot about this part.